### PR TITLE
Xfree86loader cleanup

### LINKED ISSUE #246: remove XFree86LOADER
Currently we use this define: `-UXFree86LOADER`

Which means `xf86_ansic.h` and therefore `xf86_libc.h` are never included and can be removed. To be really clean here we should backport this upstream commit: https://cgit.freedesktop.org/xorg/xserver/commit/?id=c3d14036729fd186d4ec7ca1de603e1f2d174e2f

### PR DESCRIPTION
**Pull request by sunweaver**
This PR completes the XFree86LOADER clean-up (see PR #249 and Issue #246).

Please remember to close #246 when merging this PR.

### CODE REVIEW

**Comment by sunweaver** [comment]
@uli42: please review #251. Thanks.

File: nx-X11/programs/Xserver/randr/randr.c [hunk 1]
```diff
@@ -60,6 +60,8 @@
 #include "extinit.h"
 #endif
 
+
```
Comment (uli42) [inline nx-X11/programs/Xserver/randr/randr.c]: please remove leftover empty lines

File: nx-X11/programs/Xserver/fb/fb24_32.c [hunk 2]
```diff
@@ -27,9 +27,9 @@
 #include <dix-config.h>
 #endif
 
+#include <string.h>
 #ifdef XFree86LOADER
```
Comment (uli42) [inline nx-X11/programs/Xserver/fb/fb24_32.c]: forgot to remove XFree86LOADER?

File: nx-X11/programs/Xserver/GL/glx/glxscreens.c [hunk 3]
```diff
@@ -37,12 +37,9 @@
 #include <dix-config.h>
 #endif
 
-#ifdef IN_MODULE
-#include <xf86_ansic.h>
-#else
+
```
Comment (uli42) [inline nx-X11/programs/Xserver/GL/glx/glxscreens.c]: superfluous empty line

File: nx-X11/programs/Xserver/Xext/Imakefile [hunk 4]
```diff
@@ -25,11 +25,6 @@
 
 #include <Server.tmpl>
 
-#if DoLoadableServer
-#define IHaveSubdirs
-SUBDIRS = extmod
-#endif
-
```
Comment (uli42) [inline nx-X11/programs/Xserver/Xext/Imakefile]: DoLoadableServer changes might conflict with https://github.com/uli42/nx-libs/commit/df6e43da96985e41366d06583786b64b25577939 from my PR #267

**Review by uli42 — CHANGES_REQUESTED** [review]

File: nx-X11/programs/Xserver/Xext/Imakefile (hunk 4 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/Xext/Imakefile]: I will merge #267 first and then rebase this PR.

**Comment by sunweaver** [comment]
On  So 30 Okt 2016 21:44:43 CET, Ulrich Sibiller wrote:

> uli42 commented on this pull request.
> 
> > @@ -27,9 +27,9 @@
> >  #include <dix-config.h>
> >  #endif
> 
> +#include <string.h>
>  #ifdef XFree86LOADER
> 
> Hmm, didn't you want to remove XFree86LOADER completely?

Ah. Good catch!

## Mike

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

File: nx-X11/programs/Xserver/randr/randr.c (hunk 1 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/randr/randr.c]: > please remove leftover empty lines (in randr.c)

done

File: nx-X11/programs/Xserver/fb/fb24_32.c (hunk 2 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/fb/fb24_32.c]: > forgot to remove XFree86LOADER? (in fb24_32.c)

yep. removed now.

File: nx-X11/programs/Xserver/GL/glx/glxscreens.c (hunk 3 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/GL/glx/glxscreens.c]: glxscreens.c is not part of this PR anymore

File: nx-X11/programs/Xserver/Xext/Imakefile (hunk 4 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/Xext/Imakefile]: done with rebasing...

File: nx-X11/programs/Xserver/GL/glx/glxscreens.c (hunk 3 above)
Comment (sunweaver) [reply nx-X11/programs/Xserver/GL/glx/glxscreens.c]: glxscreens.c is still _part_ of this PR... empty line remove now.

**Comment by sunweaver** [comment]
@uli42: PR #251 is ready for a second interation. Thanks!

**Comment by sunweaver** [comment]
@uli42: Did you have any time to look into PR #251 once more?

File: nx-X11/programs/Xserver/Xext/shm.c [hunk 5]
```diff
@@ -58,9 +58,7 @@ in this Software without prior written authorization from The Open Group.
 #define _XSHM_SERVER_
 #include <X11/extensions/shmstr.h>
 #include <nx-X11/Xfuncproto.h>
-#ifdef EXTMODULE
-#include "xf86_ansic.h"
-#endif
+
```
Comment (uli42) [inline nx-X11/programs/Xserver/Xext/shm.c:61]: why that extra line?

File: nx-X11/programs/Xserver/GL/glx/glxscreens.c [hunk 6]
```diff
@@ -37,12 +37,8 @@
 #include <dix-config.h>
 #endif
 
-#ifdef IN_MODULE
```
Comment (uli42) [inline nx-X11/programs/Xserver/GL/glx/glxscreens.c:40]: there are lots of other IN_MODULE checks all over the code. The should go, too!

**Review by uli42 — CHANGES_REQUESTED** [review]
Have a look at `extras/Mesa/src/mesa/main/imports.c`, I can see lots of `XFree86LOADER` stuff there

**Comment by sunweaver** [comment]
@uli42: I have reworked this PR, please take a second look, make comments and if ok, please approve. Otherwise specify change requests. Thanks!

**Review by uli42 — APPROVED** [review]
Looks good now.

**Pull request merged**